awesomplete.js

```diff
@@ -236,6 +248,25 @@ _.prototype = {
 		var value = this.input.value;
 
 		if (value.length >= this.minChars && this._list.length > 0) {
+			/* Check if we still have to delay the evaluate function. If we do, delay it by filterDelay (but we will still
+			use the latest value). We treat this function the same as an input triggered select. So if another
+			select triggers before the delay is over we might delay it again.*/
+			var time = new Date().getTime();
+
+			if (time - this.lastFilterTime < this.filterDelay) {
+				if (!this.filterDelayed) {
+					setTimeout(function () {
+							return this.evaluate();
+						}.bind(this),
+						this.filterDelay);
+					this.filterDelayed = true;
+				}
+				return;
+			}
+			this.lastFilterTime = time;
+			this.filterDelayed = false;
```

Comment:
Do we really need a separate flag for this? Wouldn't checking `this.filterDelay > 0)` suffice?